docs/user-guide/isis/sans2d.ipynb

```diff
@@ -47,97 +49,130 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "82319158",
+   "metadata": {},
+   "source": [
+    "Note the red boxes which indicate missing input parameters.\n",
+    "We can set these missing parameters, as well as parameters where we do not want to use the defaults:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4a2df47b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "workflow[OutFilename] = 'reduced.nxs'\n",
+    "\n",
+    "workflow[NeXusMonitorName[Incident]] = 'monitor2'\n",
+    "workflow[NeXusMonitorName[Transmission]] = 'monitor4'\n",
+    "\n",
+    "workflow[isis.SampleOffset] = sc.vector([0.0, 0.0, 0.053], unit='m')\n",
+    "workflow[isis.MonitorOffset[Transmission]] = sc.vector([0.0, 0.0, -6.719], unit='m')\n",
+    "\n",
+    "workflow[WavelengthBins] = sc.linspace(\n",
+    "    'wavelength', start=2.0, stop=16.0, num=141, unit='angstrom'\n",
     ")\n",
-    "providers = providers + (\n",
-    "    isis.data.transmission_from_background_run,\n",
-    "    isis.data.transmission_from_sample_run,\n",
-    "    sans.beam_center_from_center_of_mass,\n",
+    "\n",
+    "workflow[isis.sans2d.LowCountThreshold] = sc.scalar(100, unit='counts')\n",
+    "\n",
+    "mask_interval = sc.array(dims=['wavelength'], values=[2.21, 2.59], unit='angstrom')\n",
+    "workflow[WavelengthMask] = sc.DataArray(\n",
+    "    sc.array(dims=['wavelength'], values=[True]),\n",
+    "    coords={'wavelength': mask_interval},\n",
     ")\n",
     "\n",
-    "pipeline = sciline.Pipeline(providers=providers, params=params)"
+    "workflow[QBins] = sc.linspace(dim='Q', start=0.01, stop=0.6, num=141, unit='1/angstrom')\n",
+    "workflow[NonBackgroundWavelengthRange] = sc.array(\n",
+    "    dims=['wavelength'], values=[0.7, 17.1], unit='angstrom'\n",
+    ")\n",
+    "workflow[CorrectForGravity] = True\n",
+    "workflow[UncertaintyBroadcastMode] = UncertaintyBroadcastMode.upper_bound\n",
+    "workflow[ReturnEvents] = True"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "39f70669-0771-4a59-8e10-95c9120d0e9e",
+   "id": "67b18b0d",
    "metadata": {},
    "source": [
-    "## Visualize the pipeline\n",
+    "## Configuring data to load\n",
     "\n",
-    "Before we begin computations, we can visualize the pipeline:"
+    "We have not configured which files we want to load.\n",
+    "In this tutorial, we use helpers to fetch the tutorial data which return the filenames of the cached files.\n",
+    "In a real use case, you would set these parameters manually:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "d71780b8-56d5-445f-9d43-635d3d5f406b",
+   "id": "ff1658c5",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# left-right layout works better for this graph\n",
-    "pipeline.visualize(BackgroundSubtractedIofQ, graph_attr={'rankdir': 'LR'})"
+    "workflow[DirectBeamFilename] = isis.data.sans2d_tutorial_direct_beam()\n",
```

Comment:
I'm still not entirely sure about this change. I think it may be making the notebook less clear.
Our initial users will be ISs and IDSs, and I think they liked to be able to see the name of the file, because it's usually a file they gave us, and they can then know exactly what is being loaded (even if the file names were not the best).
In addition, it won't necessarily be clear to them what to do if they just want to load a local disk file.

It is also a bit annoying for us that we have to write a function for each file.

Could we make something in-between, like an object or a function that returns file paths? e.g.
```Py
finder = TutorialFileFinder()
workflow[DirectBeamFilename] = finder('DIRECT_SANS2D_REAR_34327_4m_8mm_16Feb16.dat')
workflow[Filename[SampleRun]] = finder('SANS2D00063114.nxs')
...
```
You could then have a `ScicatFileFinder` or something with a better name?

Reply:
> Could we make something in-between, like an object or a function that returns file paths?

That worked out badly so far:
- We had to have special filename mappings (adding suffix, special handling of zip files)
- It is confusing for users, since they may thing that they can use the `finder` for their own files.

> In addition, it won't necessarily be clear to them what to do if they just want to load a local disk file.

I would argue against that. It think it is much clearer now than before, and having `finder` would imho just add more confusion.

> It is also a bit annoying for us that we have to write a function for each file.

I think this is actually a positive change. It avoid hard-coding filenames in notebooks and unit tests.

Reply:
> since they may thing that they can use the finder for their own files.

True

> It avoid hard-coding filenames in notebooks and unit tests.

I agree that it means if we want to change the file used in tests and the notebooks, we only have one place to change.
However, does this mean we should avoid putting file numbers in the name of the function then? e.g. `loki_tutorial_sample_run_60339` as this would mean if we change that file we have to go and change multiple places in the codebase?
If so, we may struggle to come up with names for all these functions?

Reply:
> as this would mean if we change that file we have to go and change multiple places in the codebase?

A modern IDE can easily rename such functions (easier than finding all occurrences of a filename)?

I chose to include the run numbers in the Loki case (and only there), since the scientists are actually working with those files, and there are multiple. We can name them `sample_run_1` and `sample_run_2` if you prefer.

Reply:
I don't really mind either way.
Maybe it's not really that we need to change multiple places, but it would prevent having a function named `loki_tutorial_sample_run_60339()` whose contents get edited and it now returns the file `60777` if one forgets to rename the function.
But once again, I'm ok with either solution.

Reply:
I'll leave it then.